File: Grind_75/majorityElement.js
var majorityElement = function(numbers) {
    let obj = {};
    let result = 0;
    let majority = 0;
    for (let i = 0; i < numbers.length; i ++) {
        let key = numbers[i];
        obj[key] ? obj[key] +=1 : obj[key] = 1;
       if ( obj[key] > majority ) {
            majority = obj[key];
            result = key;
        }
    }
    return result;
};

console.log(majorityElement([2,2,1,1,1,2,2])); // 2
console.log(majorityElement([3,2,3])); // 3

File: PythonStuff/majorityElement.py
def majorityElement(numbers):
    result = 0
    majority = 0
    obj = {}

    for index, element in enumerate(numbers):
        try:
            obj[element] += 1
        except:
            obj[element] = 1
        
        if obj[element] > majority:
            majority = obj[element]
            result = element
    
    return result
    
print(majorityElement([1, 1, 1, 1, 2, 2]))
